code/parachain/frame/cosmwasm/src/lib.rs

```diff
@@ -462,6 +462,7 @@ pub mod pallet {
 			gas: u64,
 			message: ContractMessageOf<T>,
 		) -> DispatchResultWithPostInfo {
+			ensure_minimum_gas::<T>(gas, T::WeightInfo::instantiate(funds.len() as u32))?;
```

Comment:
i think it is next:
1. instantiate pallet overhead with dummy contract instance was tested
2. that is payed  by #[pallet::weight(T::WeightInfo::instantiate(funds.len()), plus gas taken at same time. 
3. so for any call here, user never ever gets T::WeightInfo::instantiate(funds.len()  back
4. so no reason to check it that is part of gas
5. instantiate ALWASY takes T::WeightInfo::instantiate(funds.len()  and never returns - in case error too.

Reply:
so tbh review from Halborn is garbage

Reply:
ensure_minimum_gas::<T>(gas, T::WeightInfo::instantiate(funds.len() as u32))?; 

may be
ensure!(gas >= T::WeightInfo::instantiate(funds.len() as u32))?;